rules/xcodeproj.bzl

```diff
@@ -652,11 +652,19 @@ def _xcodeproj_impl(ctx):
     proj_settings_debug = {
         "GCC_PREPROCESSOR_DEFINITIONS": "DEBUG",
         "SWIFT_ACTIVE_COMPILATION_CONDITIONS": "DEBUG",
+        "BAZEL_CUSTOM_BUILD_OPTIONS": ctx.attr.custom_build_options_by_config.get("Debug", []),
     }
+
+    # For release config only:
+    proj_settings_release = {
+        "BAZEL_CUSTOM_BUILD_OPTIONS": ctx.attr.custom_build_options_by_config.get("Release", []),
+    }
+
     proj_settings = {
```

Comment:
xcodeproj currently only generate debug and release builds. It will require some work to support arbitrary configs in xcodeproj. @thiagohmcruz will work on that.